src/init/features/dataconnect/index.ts

```diff
@@ -182,3 +117,154 @@ function subValues(
   }
   return replaced;
 }
+
+async function promptForService(setup: Setup, info: RequiredInfo): Promise<RequiredInfo> {
+  if (setup.projectId) {
+    await ensureApis(setup.projectId);
+    // TODO (b/344021748): Support initing with services that have existing sources/files
+    const existingServices = await listAllServices(setup.projectId);
+    const existingServicesAndSchemas = await Promise.all(
+      existingServices.map(async (s) => {
+        return {
+          service: s,
+          schema: await getSchema(s.name),
+        };
+      }),
+    );
+    const existingFreshServicesAndSchemas = existingServicesAndSchemas.filter((s) => {
+      return !s.schema?.source.files?.length;
+    });
+    if (existingFreshServicesAndSchemas.length) {
+      const choices: { name: string; value: any }[] = existingFreshServicesAndSchemas.map((s) => {
+        const serviceName = parseServiceName(s.service.name);
+        return {
+          name: `${serviceName.location}/${serviceName.serviceId}`,
+          value: s,
+        };
+      });
+      choices.push({ name: "Create a new service", value: undefined });
+      const choice: { service: Service; schema: Schema } = await promptOnce({
+        message:
+          "Your project already has existing services. Which would you like to set up local files for?",
+        type: "list",
+        choices,
+      });
+      if (choice) {
+        const serviceName = parseServiceName(choice.service.name);
+        info.serviceId = serviceName.serviceId;
+        info.locationId = serviceName.location;
+        if (choice.schema) {
+          info.cloudSqlInstanceId =
+            choice.schema.primaryDatasource.postgresql?.cloudSql.instance ?? "";
+          info.cloudSqlDatabase = choice.schema.primaryDatasource.postgresql?.database ?? "";
+        }
+      }
+    }
+  }
+
+  if (info.serviceId === "") {
+    info.serviceId = await promptOnce({
+      message: "What ID would you like to use for this service?",
+      type: "input",
+      default: "my-service",
+    });
+  }
+  return info;
+}
+
+async function promptForCloudSQLInstance(setup: Setup, info: RequiredInfo): Promise<RequiredInfo> {
+  if (setup.projectId) {
+    const instances = await cloudsql.listInstances(setup.projectId);
+    let choices = instances.map((i) => {
+      return { name: i.name, value: i.name, location: i.region };
+    });
+
+    const freeTrialInstanceId = await checkForFreeTrialInstance(setup.projectId);
+    if (!freeTrialInstanceId) {
+      choices.push({ name: "Create a new instance", value: "", location: "" });
+    }
+    // If we've already chosen a region (ie service already exists), only list instances from that region.
+    choices = choices.filter((c) => info.locationId === "" || info.locationId === c.location);
+    if (instances.length) {
```

Comment:
Very minor bug here.

```
    choices = choices.filter((c) => info.locationId === "" || info.locationId === c.location);
    if (choices.length) {
      if (!freeTrialInstanceId) {
        choices.push({ name: "Create a new instance", value: "", location: "" });
      }
      // then prompt
    }
```

If the filter list is empty, creating new instance is the only option, no need to prompt

Reply:
😓 oh, did I just trigger auto-submit?